edgepi-dac.js

```diff
@@ -28,14 +28,18 @@ module.exports = function (RED) {
       // Enforce gain on write method and check valid configuration input
       let validWriteInput;
       if(config.method === "write"){
-        dac.setDacGain(true,true);
-        const upperLim = 10;
-        const lowerLim = 0;
-        validWriteInput = checkValidWriteInput(node, voltage, upperLim, lowerLim);
+        dac.setDacGain(true,true).then(() => {
+          const upperLim = 10;
+          const lowerLim = 0;
+          validWriteInput = checkValidWriteInput(node, voltage, upperLim, lowerLim);
+          node.on('input',(msg,send,done) => inputListener(msg, send, done));
+        })
+      }
+      else {
+        node.on('input', (msg,send,done) => inputListener(msg, send, done));
       }
```

Comment:
so `dac.setDacGain` is the async function in this case? and calling .then() waits until the function is completed?

what happens when you don't wait for it.. do you get any error?

Reply:
not ATM cuz it happens so fast but it was a problem for the adc node I'm working on since that one does multiple async configurations